File: tbs220/src/main/java/edu/monash/humanise/smartcity/PayloadController.java
package edu.monash.humanise.smartcity;

import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@RequestMapping("/api/payload")
@AllArgsConstructor
public class PayloadController {
    private final PayloadService payloadService;

    @PostMapping
    public void decodeUplinkPayload(@RequestBody PayloadUplinkRequest payloadRequest) {
        log.info("New uplink payload to decode {}", payloadRequest);
        payloadService.decodeUplinkPayload(payloadRequest);
    }
}

File: athom-smart-plug/src/main/kotlin/edu/monash/humanise/smartcity/athomsmartplug/PayloadController.kt
package edu.monash.humanise.smartcity.athomsmartplug

import io.github.oshai.KotlinLogging
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

private val logger = KotlinLogging.logger {}


/**
 * Spring controller for payload POST endpoint to receive data from the collector.
 */
@RestController
@RequestMapping("/api/payload")
class PayloadController(
    private val payloadService: PayloadService,
    private val stats: Stats
) {

    /**
     * POST endpoint to receive data from the collector.
     */
    @PostMapping
    fun decodeUplinkPayload(@RequestBody payloadRequest: PayloadUplinkRequest) {
        logger.debug { "New uplink payload to decode $payloadRequest" }
        stats.incrementCounter(payloadRequest.deviceName)
        payloadService.decodeUplinkPayload(payloadRequest)
    }
}